crates/noirc_frontend/src/lexer/token.rs

```diff
@@ -316,6 +316,36 @@ impl IntType {
     }
 }
 
+/// TestScope is used to specify additional annotations for test functions
+#[derive(PartialEq, Eq, Hash, Debug, Clone, PartialOrd, Ord)]
+pub enum TestScope {
+    /// If a test has a scope of ShouldFail, then it is expected to fail
+    ShouldFail,
+    /// No scope is applied and so the test must pass
+    None,
+}
+
+impl TestScope {
+    fn from_str(string: &str) -> Option<TestScope> {
+        match string {
+            "should_fail" => Some(TestScope::ShouldFail),
+            _ => None,
+        }
+    }
```

Comment:
Should this instead implement `FromStr`?

Reply:
I thought of using FromStr but it returns a Result, whereas if this fails, it can only be for one reason. changing the name is a good idea though, so its less confusing

Reply:
We could have a Result and have the error type be `()` which is isomorphic to Option<T>